Yank/repex.py

```diff
@@ -1181,7 +1237,9 @@ class while the simulation is running. This reads the SamplerStates of every
     # Constructors.
     # -------------------------------------------------------------------------
 
-    def __init__(self, mcmc_moves=None, number_of_iterations=1, replica_mixing_scheme='swap-all'):
+    def __init__(self, mcmc_moves=None, number_of_iterations=1, replica_mixing_scheme='swap-all',
+                 online_analysis_interval=None, online_analysis_target_error=1.0,
```

Comment:
I agree. This is per-phase, so the compound error could be very large. I knew that 1kT was going to be too large, but I wanted to get this feature up as a PR for discussion and refine a bit after (like now). 

I'm still wanting to put tests in, but I cant think of how to easily do that since this relies on the `analyze` module which we have no tests for anyways because they will all be slow and require real simulation data.

I have some ideas for a very small, simple dataset which can be generated at test run-time. We would be able to use the API and just feed it really, really simple harmonic oscillator, tens of LJ fluid system, or something, make X time steps at very few states, save and analyze. We would want this instead of a static dataset since we are testing functionality and constantly changing that at the moment.